src/code_e92d0_len_5da0.c

```diff
@@ -94,20 +92,27 @@ s32 si_handle_end_if(script_context* script) {
     return 2;
 }
 
-INCLUDE_ASM(code_e92d0_len_5da0, si_handle_switch);
+s32 si_handle_switch(script_context* script) {
+    bytecode value = get_variable(script, *script->ptrReadPos);
+    s32 switchDepth = ++script->switchDepth;
 
-INCLUDE_ASM(code_e92d0_len_5da0, si_handle_switch_const);
-/*s32 si_handle_switch_const(script_context* script) {
-    s32 ptrReadPos = *script->ptrReadPos;
-    s8 switchDepth = script->switchDepth + 1;
-    
-    if (switchDepth >= 8) {
-        inf_loop: goto inf_loop; //todo
-    }
-    script->switchBlockValue[script->switchDepth + 1] = ptrReadPos;
-    script->switchBlockState[script->switchDepth + 1] = 1;
+    ASSERT(switchDepth < 8);
+
+    script->switchBlockValue[switchDepth] = value;
+    script->switchBlockState[switchDepth] = 1;
     return 2;
-}*/
+}
+
+s32 si_handle_switch_const(script_context* script) {
+    bytecode value = *script->ptrReadPos;
+    s32 switchDepth = ++script->switchDepth;
```

Comment:
can we do post-increment unless otherwise necessary?

Reply:
It's necessary! Alternatively we could do
```c
script->switchDepth++;
s32 switchDepth = switchDepth;
```
but I don't like that as much.

Reply:
Oh I see why that's different - yeah. Okay, carry on :)